clash-prelude/src/Clash/Sized/Internal/BitVector.hs

```diff
@@ -500,6 +500,9 @@ maxBound# :: forall n . KnownNat n => BitVector n
 maxBound# = let m = 1 `shiftL` fromInteger (natVal (Proxy @n))
             in  BV 0 (m-1)
 
+-- | __NB__: `fromInteger` can cause unexpected truncation, as `Integer` is
+-- arbitrarily bounded during synthesis.  Prefer
+-- `Clash.Class.BitPack.bitCoerce` and the `Resize` class.
```

Comment:
```suggestion
-- | __NB__: 'fromInteger' can cause unexpected truncation, as 'Integer' is
-- arbitrarily bounded during synthesis.  Prefer
-- 'Clash.Class.BitPack.bitCoerce' and the 'Resize' class.
```

Reply:
If you use this Haddock can link it in the documentation.

Reply:
Whoops!  Been writing too much markdown :laughing:

Reply:
I know the feeling! It is so difficult switching between the two.